hotel_elasticsearch/backup.py

```diff
@@ -1,36 +1,193 @@
+import datetime
+import os
 import random
 import time
-import datetime
-from hotel_elasticsearch.dist_locks import Timer, BlockedByAnotherTimerException
+
 import requests
 from requests.exceptions import HTTPError
 import logging
 
+from hotel_elasticsearch.alerting import alerter_factory
+from hotel_elasticsearch.configuration import HotelElasticSearchConfig
+
 logger = logging.getLogger('hotel_elasticsearch.backup')
 
+
+class BackupException(Exception):
+    pass
+
+
 class BackupManager(object):
-    def run_backup(self):
-        now = datetime.datetime.now()
-        snapshot_name = 'snapshot_%s-%s-%s_%s' % (now.year, now.month, now.day, now.hour)
-        logger.info("Executing backup command, snapshot name: %s" % snapshot_name)
-        r = requests.put("http://localhost:9200/_snapshot/%s/%s/" % (
-            'hotel-backup',
-            snapshot_name
-        ))
-        r.raise_for_status()
-        logger.info("Snapshot successfully scheduled")
-
-
-def backup_thread():
-    while True:
+    def __init__(self, cluster_node):
+        self.cluster_node = cluster_node
+        self.config = HotelElasticSearchConfig()
+
+    def check_backup(self):
+        # Only the elected master should check the backup
+        if not self.cluster_node.is_elected_master:
+            return
+        try:
+            try:
+                self._check_backup_configuration()
+            except HTTPError as e:
+                logger.exception(e)
+                self.configure_backup()
+
+            try:
+                result = requests.get('http://localhost:9200/_snapshot/cluster_backup/*?order=desc&size=1')
+                result.raise_for_status()
+            except HTTPError as e:
+                logger.exception(e)
+                raise BackupException('Backup listing request failed')
+
+            result_dict = result.json()
+            if 'snapshots' not in result_dict or len(result_dict['snapshots']) == 0:
+                raise BackupException('No backups found')
+            else:
+                snapshot = result_dict['snapshots'][0]
+                if snapshot['state'] not in ['SUCCESS', 'IN_PROGRESS']:
+                    raise BackupException('Latest backup is not in a success state')
+                elif snapshot["start_time"] < datetime.datetime.now() - datetime.timedelta(days=1):
+                    raise BackupException('Latest backup is older than 1 day')
+        except BackupException as e:
```

Comment:
Is this try except only used for control flow?

Reply:
I don't think so, it doesn't control any order of execution. It's there for alerting purposes